### PR TITLE
fix: add eslint and settings from protocol

### PR DESCRIPTION
**Pull request by daywiss**
override tsdx linitng with eslint and rules from protocol linter.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,15 +1,31 @@
 module.exports = {
-  "extends": [
-    "react-app",
-    "prettier/@typescript-eslint",
+  env: {
```
Comment (mrice32) [inline .eslintrc.js:2]: Are these just copied from the protocol repo?

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm test
```
Comment (mrice32) [inline .husky/pre-commit:4]: You want to run all tests before each commit?

File: package.json [hunk 3]
```diff
@@ -18,15 +18,15 @@
     "test": "tsdx test",
     "test:watch": "tsdx test --watch",
     "test:e2e": "tsdx test --config jest-e2e-config.json",
-    "lint": "tsdx lint --fix",
-    "prepare": "tsdx build",
+    "lint": "eslint --fix src",
```
Comment (mrice32) [inline package.json:21]: Should we be running prettier as well? Or just eslint?

Comment (mrice32) [inline package.json:21]: Do you want to have ci check that the code was linted correctly? Like a lint-check command?

**Review by mrice32 — APPROVED** [review]
Looks great!

File: .eslintrc.js (hunk 1 above)
Comment (daywiss) [reply .eslintrc.js:2]: it mostly was, i didnt need to add overrides and removed one rule about flattening brackets which was causing warnings.

File: .husky/pre-commit (hunk 2 above)
Comment (daywiss) [reply .husky/pre-commit:4]: yea im ok with that, what do u think?

File: package.json (hunk 3 above)
Comment (daywiss) [reply package.json:21]: i think that would be useful, i could add that

**Review by amateima — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (daywiss) [reply package.json:21]: i added `lint-check`

Comment (daywiss) [reply package.json:21]: i added `lint-check`

**Pull request merged**